volk_32f_s32f_s32f_mod_range_32f.h doesn't compile with MSVC

The errors can be seen here:
[05-Build GNURadio Release.txt](https://github.com/gnuradio/volk/files/1685321/05-Build.GNURadio.Release.txt)

MSVC compiler basically complains about lack of two intrinsics:
_mm_cvttps_pi32
_mm_cvtpi32x2_ps

This is funny as both of them are defined in xmmintrin.h that is included. In MSCV's xmmintrin.h they are inside of #if defined(_M_IX86) statement, but defining _M_IX86 manually didn't help.

@marcusmueller I don't have good news yet. After trying to compile that branch I get different error message:
```
83>C:\gr-build\src-stage3\src\gnuradio\volk\kernels\volk/volk_32f_s32f_s32f_mod_range_32f.h(308): error C2679: binary '=': no operator found which takes a right-hand operand of type '__m128i' (or there is no acceptable conversion) (compiling source file C:\gr-build\src-stage3\build\Release\volk\lib\volk_machine_sse2.c) [C:\gr-build\src-stage3\build\Release\volk\lib\volk_obj.vcxproj]
         C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xmmintrin.h(75): note: could be '__m128 &__m128::operator =(__m128 &&)' (compiling source file C:\gr-build\src-stage3\build\Release\volk\lib\volk_machine_sse2.c)
         C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xmmintrin.h(75): note: or       '__m128 &__m128::operator =(const __m128 &)' (compiling source file C:\gr-build\src-stage3\build\Release\volk\lib\volk_machine_sse2.c)
         C:\gr-build\src-stage3\src\gnuradio\volk\kernels\volk/volk_32f_s32f_s32f_mod_range_32f.h(308): note: while trying to match the argument list '(__m128, __m128i)' (compiling source file C:\gr-build\src-stage3\build\Release\volk\lib\volk_machine_sse2.c)
    83>C:\gr-build\src-stage3\src\gnuradio\volk\kernels\volk/volk_32f_s32f_s32f_mod_range_32f.h(309): error C2664: '__m128 _mm_cvtepi32_ps(__m128i)': cannot convert argument 1 from '__m128' to '__m128i' (compiling source file C:\gr-build\src-stage3\build\Release\volk\lib\volk_machine_sse2.c) [C:\gr-build\src-stage3\build\Release\volk\lib\volk_obj.vcxproj]
         C:\gr-build\src-stage3\src\gnuradio\volk\kernels\volk/volk_32f_s32f_s32f_mod_range_32f.h(309): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called (compiling source file C:\gr-build\src-stage3\build\Release\volk\lib\volk_machine_sse2.c)
    83>C:\gr-build\src-stage3\src\gnuradio\volk\kernels\volk/volk_32f_s32f_s32f_mod_range_32f.h(364): error C2679: binary '=': no operator found which takes a right-hand operand of type '__m128i' (or there is no acceptable conversion) (compiling source file C:\gr-build\src-stage3\build\Release\volk\lib\volk_machine_sse2.c) [C:\gr-build\src-stage3\build\Release\volk\lib\volk_obj.vcxproj]
         C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xmmintrin.h(75): note: could be '__m128 &__m128::operator =(__m128 &&)' (compiling source file C:\gr-build\src-stage3\build\Release\volk\lib\volk_machine_sse2.c)
         C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xmmintrin.h(75): note: or       '__m128 &__m128::operator =(const __m128 &)' (compiling source file C:\gr-build\src-stage3\build\Release\volk\lib\volk_machine_sse2.c)
         C:\gr-build\src-stage3\src\gnuradio\volk\kernels\volk/volk_32f_s32f_s32f_mod_range_32f.h(364): note: while trying to match the argument list '(__m128, __m128i)' (compiling source file C:\gr-build\src-stage3\build\Release\volk\lib\volk_machine_sse2.c)
    83>C:\gr-build\src-stage3\src\gnuradio\volk\kernels\volk/volk_32f_s32f_s32f_mod_range_32f.h(365): error C2664: '__m128 _mm_cvtepi32_ps(__m128i)': cannot convert argument 1 from '__m128' to '__m128i' (compiling source file C:\gr-build\src-stage3\build\Release\volk\lib\volk_machine_sse2.c) [C:\gr-build\src-stage3\build\Release\volk\lib\volk_obj.vcxproj]
         C:\gr-build\src-stage3\src\gnuradio\volk\kernels\volk/volk_32f_s32f_s32f_mod_range_32f.h(365): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called (compiling source file C:\gr-build\src-stage3\build\Release\volk\lib\volk_machine_sse2.c)
   362>CustomBuild:
```

The full build log is here:
[05-Build GNURadio Release.txt](https://github.com/gnuradio/volk/files/1726941/05-Build.GNURadio.Release.txt)

--- a/kernels/volk/volk_32f_s32f_s32f_mod_range_32f.h
+++ b/kernels/volk/volk_32f_s32f_s32f_mod_range_32f.h
@@ -289,7 +289,6 @@ static inline void volk_32f_s32f_s32f_mod_range_32f_u_sse(float* outputVector, c
   __m128 input, output;
   __m128 is_smaller, is_bigger;
   __m128 excess, adj;
-  __m64 lo, hi;
 
   const float *inPtr = inputVector;
   float *outPtr = outputVector;
@@ -305,13 +304,9 @@ static inline void volk_32f_s32f_s32f_mod_range_32f_u_sse(float* outputVector, c
     excess = _mm_or_ps(_mm_and_ps(_mm_sub_ps(input, upper), is_bigger), excess);
     // how many do we have to add? (int(excess/distance+1)*distance)
     excess = _mm_div_ps(excess, distance);
-    // round down – for some reason, SSE doesn't come with a 4x float -> 4x int32 conversion.
-    // we're using adj as temp variable
-    adj = _mm_setzero_ps();
-    adj = _mm_movehl_ps(adj, excess);
-    lo = _mm_cvttps_pi32(excess);
-    hi = _mm_cvttps_pi32(adj);
-    excess = _mm_cvtpi32x2_ps(lo,hi);
+    // round down – for some reason
+    excess = _mm_cvttps_epi32(excess);
+    excess = _mm_cvtepi32_ps(excess);
     // plus 1
     adj = _mm_set_ps1(1.0f);
     excess = _mm_add_ps(excess, adj);
@@ -350,7 +345,6 @@ static inline void volk_32f_s32f_s32f_mod_range_32f_a_sse(float* outputVector, c
   __m128 input, output;
   __m128 is_smaller, is_bigger;
   __m128 excess, adj;
-  __m64 lo, hi;
 
   const float *inPtr = inputVector;
   float *outPtr = outputVector;
@@ -366,13 +360,9 @@ static inline void volk_32f_s32f_s32f_mod_range_32f_a_sse(float* outputVector, c
     excess = _mm_or_ps(_mm_and_ps(_mm_sub_ps(input, upper), is_bigger), excess);
     // how many do we have to add? (int(excess/distance+1)*distance)
     excess = _mm_div_ps(excess, distance);
-    // round down – for some reason, SSE doesn't come with a 4x float -> 4x int32 conversion.
-    // we're using adj as temp variable
-    adj = _mm_setzero_ps();
-    adj = _mm_movehl_ps(adj, excess);
-    lo = _mm_cvttps_pi32(excess);
-    hi = _mm_cvttps_pi32(adj);
-    excess = _mm_cvtpi32x2_ps(lo,hi);
+    // round down
+    excess = _mm_cvttps_epi32(excess);
+    excess = _mm_cvtepi32_ps(excess);
     // plus 1
     adj = _mm_set_ps1(1.0f);
     excess = _mm_add_ps(excess, adj);

--- a/kernels/volk/volk_32f_s32f_s32f_mod_range_32f.h
+++ b/kernels/volk/volk_32f_s32f_s32f_mod_range_32f.h
@@ -307,7 +307,7 @@ static inline void volk_32f_s32f_s32f_mod_range_32f_u_sse(float* outputVector, c
     excess = _mm_div_ps(excess, distance);
     // round down – for some reason
     rounddown = _mm_cvttps_epi32(excess);
-    excess = _mm_cvtepi32_ps(excess);
+    excess = _mm_cvtepi32_ps(rounddown);
     // plus 1
     adj = _mm_set_ps1(1.0f);
     excess = _mm_add_ps(excess, adj);
@@ -364,7 +364,7 @@ static inline void volk_32f_s32f_s32f_mod_range_32f_a_sse(float* outputVector, c
     excess = _mm_div_ps(excess, distance);
     // round down
     rounddown = _mm_cvttps_epi32(excess);
-    excess = _mm_cvtepi32_ps(excess);
+    excess = _mm_cvtepi32_ps(rounddown);
     // plus 1
     adj = _mm_set_ps1(1.0f);
     excess = _mm_add_ps(excess, adj);

--- a/kernels/volk/volk_32f_s32f_s32f_mod_range_32f.h
+++ b/kernels/volk/volk_32f_s32f_s32f_mod_range_32f.h
@@ -289,6 +289,7 @@ static inline void volk_32f_s32f_s32f_mod_range_32f_u_sse(float* outputVector, c
   __m128 input, output;
   __m128 is_smaller, is_bigger;
   __m128 excess, adj;
+  __m128i rounddown;
 
   const float *inPtr = inputVector;
   float *outPtr = outputVector;
@@ -305,7 +306,7 @@ static inline void volk_32f_s32f_s32f_mod_range_32f_u_sse(float* outputVector, c
     // how many do we have to add? (int(excess/distance+1)*distance)
     excess = _mm_div_ps(excess, distance);
     // round down – for some reason
-    excess = _mm_cvttps_epi32(excess);
+    rounddown = _mm_cvttps_epi32(excess);
     excess = _mm_cvtepi32_ps(excess);
     // plus 1
     adj = _mm_set_ps1(1.0f);
@@ -345,6 +346,7 @@ static inline void volk_32f_s32f_s32f_mod_range_32f_a_sse(float* outputVector, c
   __m128 input, output;
   __m128 is_smaller, is_bigger;
   __m128 excess, adj;
+  __m128i rounddown;
 
   const float *inPtr = inputVector;
   float *outPtr = outputVector;
@@ -361,7 +363,7 @@ static inline void volk_32f_s32f_s32f_mod_range_32f_a_sse(float* outputVector, c
     // how many do we have to add? (int(excess/distance+1)*distance)
     excess = _mm_div_ps(excess, distance);
     // round down
-    excess = _mm_cvttps_epi32(excess);
+    rounddown = _mm_cvttps_epi32(excess);
     excess = _mm_cvtepi32_ps(excess);
     // plus 1
     adj = _mm_set_ps1(1.0f);

